contribs/analysis/src/main/java/org/matsim/contrib/analysis/vsp/traveltimedistance/TravelTimeValidationRunnerPreAnalysis.java

```diff
@@ -0,0 +1,126 @@
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+
+public class TravelTimeValidationRunnerPreAnalysis {
+    private final Network network;
+    private final int trips;
+    private Geometry keyAreaGeometry;
+    private final TravelTimeDistanceValidator validator;
+    private final String outputFolder;
+
+    private final Random rnd = new Random(1234);
+
+    public TravelTimeValidationRunnerPreAnalysis(Network network, int trips, String outputFolder, TravelTimeDistanceValidator validator, Geometry keyAreaGeometry) {
+        this.network = network;
+        this.trips = trips;
+        this.outputFolder = outputFolder;
+        this.validator = validator;
+        this.keyAreaGeometry = keyAreaGeometry;
+    }
+
+    public void setKeyAreaGeometry(Geometry keyAreaGeometry) {
+        this.keyAreaGeometry = keyAreaGeometry;
+    }
+
+    public void run() throws IOException {
+        List<Link> links = network.getLinks().values().stream().
+                filter(l -> l.getAllowedModes().contains(TransportMode.car)).
+                collect(Collectors.toList());
+        int numOfLinks = links.size();
+
+        List<Link> linksInsideShp = new ArrayList<>();
+        List<Link> outsideLinks = new ArrayList<>();
+
+        if (keyAreaGeometry != null) {
+            for (Link link : links) {
+                if (MGC.coord2Point(link.getToNode().getCoord()).within(keyAreaGeometry)) {
+                    linksInsideShp.add(link);
+                }
+            }
+            outsideLinks.addAll(links);
+            outsideLinks.removeAll(linksInsideShp);
+        }
+
+        // Create router
+        FreeSpeedTravelTime travelTime = new FreeSpeedTravelTime();
+        LeastCostPathCalculatorFactory fastAStarLandmarksFactory = new SpeedyALTFactory();
+        OnlyTimeDependentTravelDisutilityFactory disutilityFactory = new OnlyTimeDependentTravelDisutilityFactory();
+        TravelDisutility travelDisutility = disutilityFactory.createTravelDisutility(travelTime);
+        LeastCostPathCalculator router = fastAStarLandmarksFactory.createPathCalculator(network, travelDisutility,
+                travelTime);
+
+        // Choose random trips to validate
+        CSVPrinter csvWriter = new CSVPrinter(new FileWriter(outputFolder + "/pre-analysis-results.csv"), CSVFormat.DEFAULT);
+        csvWriter.printRecord("trip_number", "trip_category", "from_x", "from_y", "to_x", "to_y", "simulated_travel_time", "validated_travel_time");
+        int counter = 0;
+
+        Link fromLink;
+        Link toLink;
+        String tripType;
+        while (counter < trips) {
+            if (!linksInsideShp.isEmpty()) {
+                int numOfLinksInsideShp = linksInsideShp.size();
+                int numOfOutsideLinks = outsideLinks.size();
+
+                if (counter < 0.6 * trips) {
```

Comment:
What are these magic numbers 0.6 an 0.9? Maybe they should be configurable?

Reply:
At least they should be a static final variable with a short description.

Reply:
Yes, those can be configuratble. But for now I just add it in to avoid too many input arguments. So I will move it to the field of the class and add description.